java/daikon/DynComp.java

```diff
@@ -122,6 +142,38 @@ public class DynComp {
   /** Synopsis for the DynComp command line. */
   public static final String synopsis = "daikon.DynComp [options] target [target-args]";
 
+  /**
+   * Locate the classfile file that a class was loaded from. The result is a URL for the file
+   * (starting with "file:" or "jar:file:").
+   *
+   * @param c class to look up
+   * @return URL to the classfile, or null if not found
+   */
+  public static @Nullable String whereLoadedFrom(Class<?> c) {
+    if (c == null) {
```

Comment:
Similar to above: If this is a general utility method we need to check c before calling c.getClassLoader().  Again, this cannot happen in our case.  Do we want to keep general code or custom tailor to our use?

Reply:
If this is a general utility method, its specification could say that the argument cannot be null.  Then the check wouldn't be needed.
Currently, the specification doesn't say what happens for null.  Is null being the argument is the only situation where null is the result?  If so, it would simplify the specification to forbid null as an argument.
More generally, I don't see how permitting null as an argument helps clients.